### ISSUE TITLE
False positives for devDependencies used in scripts

### ISSUE THREAD

**Issue by TheSimpleZ** [depcheck/depcheck#630]
When running depcheck i get the following false positives:

Unused devDependencies
* nodemon
* eslint
* eslint-plugin-import
* eslint-plugin-mocha
* license-checker
* mocha

All of them, except the eslint plugins are used in the package.json

<details>
<summary>package.json</summary>

```json
{
  "name": "authentication",
  "version": "0.6.4",
  "description": "An authentication service using Active Directory.",
  "main": "src/server.js",
  "license": "SEE LICENSE IN LICENSE",
  "scripts": {
    "lint:fix": "eslint -c .eslint.json  --no-color --fix .",
    "lint": "eslint -c .eslint.json --no-color .",
    "start": "nodemon src/server.js",
    "test": "mocha --recursive --exit ./src/tests",
    "license": "license-checker"
  },
  "dependencies": {
    "@google-cloud/logging-winston": "^4.0.4",
    "@google-cloud/secret-manager": "^3.6.0",
    "@ingka-csr/opentelemetry": "^4.0.2",
    "body-parser": "^1.19.0",
    "config": "^3.3.1",
    "express": "^4.17.1",
    "jsonwebtoken": "^8.5.1",
    "jwks-rsa": "^1.8.0",
    "lodash": "^4.17.20",
    "passport": "^0.4.1",
    "passport-azure-ad": "^4.2.1",
    "request-promise": "^4.2.6",
    "winston": "^3.2.1"
  },
  "devDependencies": {
    "nodemon": "^2.0.3",
    "chai": "4.3.4",
    "chai-as-promised": "7.1.1",
    "chai-http": "4.3.0",
    "eslint": "7.22.0",
    "eslint-config-airbnb-base": "14.2.1",
    "eslint-plugin-import": "2.22.1",
    "eslint-plugin-mocha": "8.1.0",
    "license-checker": "25.0.1",
    "mocha": "8.3.2",
    "mock-require": "3.0.3",
    "nock": "13.0.7",
    "sinon": "10.0.0"
  }
}
```
</details>

<details>
<summary>$ npx depcheck box-authentication/code --json | jq .</summary>

```json
{
  "dependencies": [],
  "devDependencies": [
    "nodemon",
    "eslint",
    "eslint-plugin-import",
    "eslint-plugin-mocha",
    "license-checker",
    "mocha"
  ],
  "missing": {},
  "using": {
    "eslint-config-airbnb-base": [
      "/home/zrean/projects/returns/box-authentication/code/package.json"
    ],
    "body-parser": [
      "/home/zrean/projects/returns/box-authentication/code/src/app.js"
    ],
    "request-promise": [
      "/home/zrean/projects/returns/box-authentication/code/src/getAuthToken.js"
    ],
    "@ingka-csr/opentelemetry": [
      "/home/zrean/projects/returns/box-authentication/code/src/server.js"
    ],
    "express": [
      "/home/zrean/projects/returns/box-authentication/code/src/server.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/app.test.js"
    ],
    "chai": [
      "/home/zrean/projects/returns/box-authentication/code/src/tests/app.test.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/getAuthToken.test.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/strategies/application-strategy.test.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/libs/config.production.test.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/libs/config.test.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/libs/utils.test.js"
    ],
    "chai-http": [
      "/home/zrean/projects/returns/box-authentication/code/src/tests/app.test.js"
    ],
    "mock-require": [
      "/home/zrean/projects/returns/box-authentication/code/src/tests/app.test.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/libs/config.production.test.js"
    ],
    "sinon": [
      "/home/zrean/projects/returns/box-authentication/code/src/tests/app.test.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/libs/config.production.test.js"
    ],
    "nock": [
      "/home/zrean/projects/returns/box-authentication/code/src/tests/getAuthToken.test.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/strategies/application-strategy.test.js"
    ],
    "chai-as-promised": [
      "/home/zrean/projects/returns/box-authentication/code/src/tests/strategies/application-strategy.test.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/libs/config.production.test.js",
      "/home/zrean/projects/returns/box-authentication/code/src/tests/libs/config.test.js"
    ],
    "passport-azure-ad": [
      "/home/zrean/projects/returns/box-authentication/code/src/strategies/application-strategy.js"
    ],
    "jsonwebtoken": [
      "/home/zrean/projects/returns/box-authentication/code/src/strategies/application-strategy.js"
    ],
    "passport": [
      "/home/zrean/projects/returns/box-authentication/code/src/strategies/application-strategy.js"
    ],
    "jwks-rsa": [
      "/home/zrean/projects/returns/box-authentication/code/src/strategies/application-strategy.js"
    ],
    "config": [
      "/home/zrean/projects/returns/box-authentication/code/src/libs/config.js"
    ],
    "lodash": [
      "/home/zrean/projects/returns/box-authentication/code/src/libs/config.js"
    ],
    "@google-cloud/logging-winston": [
      "/home/zrean/projects/returns/box-authentication/code/src/libs/logger.js"
    ],
    "winston": [
      "/home/zrean/projects/returns/box-authentication/code/src/libs/logger.js"
    ],
    "@google-cloud/secret-manager": [
      "/home/zrean/projects/returns/box-authentication/code/src/libs/secrets.js"
    ]
  },
  "invalidFiles": {},
  "invalidDirs": {}
}
```
</details>

**Comment by rumpl** [comment]
Hum, that's odd, IIRC we are looking at `"scripts"` in the package.json file... Thanks for openning the issue. 👍

**Comment by lkashef** [comment]
Same here for: `cors`, `body-parser`, `helmet`, `@types/cors` (not imported but other types installed do not show errors)

Import statements (typescript)
```
import cors from 'cors';
import helmet from 'helmet';
import { json as parseJson, urlencoded as parseUrlencoded } from 'body-parser';
```

I run the module using package.json script (only local install, installed and running using yarn)

`"depcheck": "depcheck ./ --ignores ts-node"`

---

Happy to open another issue if you decide that this is not-related to the original issue;

**Comment by armandorod2000** [comment]
Same here... I have **ts-node** in my _DevDependencies_ and I'm using **ts-node/register** in one of my scripts (in package.json). Just like this: `"start": "...node -r ts-node/register -- ./src/app.ts"`
When I run **depcheck**, it shows **ts-node** listed as an unused dependency

**Comment by shian15810** [comment]
Not stale, issue is still persisting

**Comment by rubengmurray** [comment]
I don't get the issue with `eslint` in `devDependencies` with this included in my `package.json`:

```json
 { 
  "scripts": {
    ...,
    "lint": "eslint --fix './src/**/*.ts'",
  }
}
 ```
